src/sinol_make/util.py

```diff
@@ -154,6 +157,69 @@ def save_config(config):
             yaml.dump(config, config_file)
 
 
+def check_for_updates(current_version) -> str | None:
+    """
+    Function to check if there is a new version of sinol-make.
+    :param current_version: current version of sinol-make
+    :return: returns new version if there is one, None otherwise
+    """
+    data_dir = importlib.resources.files("sinol_make").joinpath("data")
+    if not data_dir.is_dir():
+        os.mkdir(data_dir)
+
+    thread = threading.Thread(target=check_version)
+    thread.start()
+    version_file = data_dir.joinpath("version")
+    if version_file.is_file():
+        version = version_file.read_text()
+        try:
+            if compare_versions(current_version, version) == -1:
```

Comment:
What if we just updated the version? This file probably will still exist (the `check_version` won't be fast enough to overwrite here) and the warning will be printed, even though we have just updated the version.

Reply:
Then the current version will be the same (or higher) than the one in the file, so I don't think there will be a problem

Reply:
I see, you are right. Nevertheless, adding comments explaining why this method is correct would improve readability and maintainability.